bayesian/license_extractor.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""License extractor utility functions."""
+
+from .default_config import LIC_SYNONYMS_URL
+from requests import get
+from collections import defaultdict
+from lru import lru_cache_function
+from flask import current_app
+from datetime import datetime as dt
+
+
+# caching response for 24 hours
+@lru_cache_function(max_size=2048, expiration=60 * 60 * 24)
+def get_license_synonyms():
+    """Fetch all the license sysnonyms from license anlysis github repo."""
+    resp = get(LIC_SYNONYMS_URL)
+    if resp.status_code == 200:
+        current_app.logger.info(
+            "{} Succefully fetched license synonyms".format(dt.now()))
+        return resp.json()
+    else:
+        current_app.logger.error("{tm} Unable to fetch license synonyms, STATUS_CODE:{cd}".format(
+            tm=dt.now(), cd=resp.status_code))
+        return {}
```

Comment:
Returning empty dictionary here means that it will stay cached for the whole day. Another possible approach could be to throw an exception here and thus skip caching when the GitHub request above failed from some reason.

Reply:
Thanks, for pointing that out I will do the necessary changes.